src/backend/services/chat.py

```diff
@@ -78,31 +80,46 @@ def process_chat(
     """
     user_id = get_header_user_id(request)
     deployment_name = request.headers.get("Deployment-Name", "")
-    model_config = {}
+    deployment_config = {}
     # Deployment config is the settings for the model deployment per request
     # It is a string of key value pairs separated by semicolons
     # For example: "azure_key1=value1;azure_key2=value2"
+    # TODO Eugene: Confirm it with Scott - header deployment config has priority over the deployment config in the DB,
+    # but agent deployment config has priority over both if it is set
```

Comment:
yep, i think that makes sense 👍

Reply:
actually another thought:

i think we can stop sending the deployment config in the header all together. it can be set once during agent creation and we always just check the one in the DB.